In [2]:
import os
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import plotly.express as px
from plotly.subplots import make_subplots
import plotly.io as pio
pio.renderers.defaule = 'colab'

import kagglehub

## 8. **In a NEW NOTEBOOK** Write code that reads in your subset of the data, markdown that explains clearly where you got the data originally (license and references included) and the process you took to create your subset, a description of the question you are answering, and code that can reliably run and answer your question followed by words that explain your results.

In [12]:
file = 'df_for_question.csv'
read_in_df = pd.read_csv(file)
read_in_df

Calories_Burned Workout_Type  Session_Duration (hours)
0               712.0     Strength                      1.12
1               833.0     Strength                      1.00
2              1678.0       Cardio                      1.24
3               628.0       Cardio                      1.45
4              1286.0     Strength                      1.60
...               ...          ...                       ...
1795           1388.0         HIIT                      1.11
1796           1367.0         HIIT                      0.77
1797           1261.0     Strength                      1.20
1798            956.0       Cardio                      1.04
1799           1178.0     Strength                      1.29

[1800 rows x 3 columns]

This data was obtained from Kaggle. The owner of the data goes by the name of Nadeem Majeed. The license that is on this data is titled "Community Data License Agreement - Sharing - Version 1.0". This is the same data that was used in the previous questions however this is more masked to only three columns.

Creating the subset of data was simple. I created a new variable named df_for_question and made it equal to the data frame. This data frame has only the columns calories_burned, workout_type, and Session_Duration(hours)'. This subset of data is going to be used to answer the question "What are the most effect types of workouts that will burn the most burn the most calories in the least amount of time?"

In [85]:
workout_summary = read_in_df.groupby('Workout_Type')['Calories_Burned'].mean()
workout_summary

Workout_Type
Cardio        1045.293427
HIIT          1005.000000
Strength      1019.240088
Yoga          1066.334129
\nStrength     912.916667
\tCardio      1143.454545
\tYoga        1115.000000
Name: Calories_Burned, dtype: float64

 This code is creating a variable workout_summary and is using the data frame and grouping it buy the workout type and masking the calorie burned. Through this we are able to find the average amount of calories burned for each workout. 

In [92]:
The_top = workout_summary.sort_values(ascending = False).head(2)
The_top

Workout_Type
\tCardio    1143.454545
\tYoga      1115.000000
Name: Calories_Burned, dtype: float64

We are then able to sort the values in descending order to find the top two results that burned the most calories. 

In [79]:
read_in_df['Calories_per_Hour'] = (read_in_df['Calories_Burned']/ read_in_df['Session_Duration (hours)'])

burn_rate_summary = read_in_df.groupby('Workout_Type')['Calories_per_Hour'].mean()

burn_rate_summary

Workout_Type
Cardio        837.632114
HIIT          787.695150
Strength      790.518912
Yoga          809.456354
\nStrength    701.137821
\tCardio      797.710475
\tYoga        906.471420
Name: Calories_per_Hour, dtype: float64

This code is creating a new column in the data frame named calories_per_hour and it is dividing the calories burned by the session duration hours. After we create the burn rate summary that groups the workout type and calories per hour average.

In [81]:
burn_rate_summary.sort_values(ascending = False).head(2)

Workout_Type
\tYoga    906.471420
Cardio    837.632114
Name: Calories_per_Hour, dtype: float64

In this data we are able to see that for each hour on average, yoga burned 906 calories and cardio burns 837